src/dst/src/Distributed.cc

```diff
@@ -91,13 +91,16 @@ void Distributed::runWorker(const char* ip,
   }
 }
 
-void Distributed::runLoadBalancer(const char* ip, unsigned short port)
+void Distributed::runLoadBalancer(const char* ip,
+                                  unsigned short port,
+                                  const char* workers_domain)
 {
   try {
     asio::io_service io_service;
-    LoadBalancer balancer(this, io_service, logger_, ip, port);
-    for (auto worker : end_points_)
-      balancer.addWorker(worker.ip, worker.port);
+    LoadBalancer balancer(this, io_service, logger_, ip, workers_domain, port);
+    if (std::strcmp(workers_domain, "") != 0)
```

Comment:
==

Reply:
the equality holds here but not in LoadBalancer.cc.
Here you are checking if workers_domain == "" to add workers manually from the list of workers. However, in LoadBalancer.cc you are checking that workers_domain != "" to lookUpWorkers. It's always better to use strings than cstrings to avoid such problems